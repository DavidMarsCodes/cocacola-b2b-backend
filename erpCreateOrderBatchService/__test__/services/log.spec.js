const log = require('../../services/log');

describe('Create Function', () => {

  it('You should create a log with status OK.', async () => {

    const OrderLog = { create: jest.fn().mockResolvedValue(true) };

    const order = {
      cpgId: '001',
      organizationId: '3043',
      countryId: 'AR',
      orderId: 16,
    };

    const event = 'OK';

    await log.create(OrderLog, order, event);

    expect(OrderLog.create.mock.calls).toMatchSnapshot([
      [
        {
          cpgId: '001',
          eventDatetime: expect.any(String),
          organizationId: '3043',
          countryId: 'AR',
          eventType: 'RETRY',
          eventResult: 'OK',
          eventInfo: `The operation has been successful. The order data was sent with id 16.`,
          orderId: 16,
        },
      ],
    ]);

  });

  it('You should create a log with status ERROR.', async () => {

    const OrderLog = { create: jest.fn().mockResolvedValue(true) };

    const order = {
      cpgId: '001',
      organizationId: '3043',
      countryId: 'AR',
      orderId: 16,
    };

    const error = { message: 'Test Error' };

    const event = 'ERROR';

    await log.create(OrderLog, order, event, error);

    expect(OrderLog.create.mock.calls).toMatchSnapshot([
      [
        {
          cpgId: '001',
          eventDatetime: expect.any(String),
          organizationId: '3043',
          countryId: 'AR',
          eventType: 'RETRY',
          eventResult: 'ERROR',
          eventInfo: 'Test Error',
          orderId: 16,
        },
      ],
    ]);

  });

  it('You should create a log with status ERROR and the message property must contain an empty value in case it is not an axios error.', async () => {

    const OrderLog = { create: jest.fn().mockResolvedValue(true) };

    const order = {
      cpgId: '001',
      organizationId: '3043',
      countryId: 'AR',
      orderId: 16,
    };

    const error = { message: 'Test Error' };

    const event = 'ERROR';

    await log.create(OrderLog, order, event);

    expect(OrderLog.create.mock.calls).toMatchSnapshot([
      [
        {
          cpgId: '001',
          eventDatetime: expect.any(String),
          organizationId: '3043',
          countryId: 'AR',
          eventType: 'RETRY',
          eventResult: 'ERROR',
          eventInfo: '',
          orderId: 16,
        },
      ],
    ]);

  });

});