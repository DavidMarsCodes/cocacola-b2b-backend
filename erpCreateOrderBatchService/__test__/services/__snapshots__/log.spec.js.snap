// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Function You should create a log with status ERROR and the message property must contain an empty value in case it is not an axios error. 1`] = `
Array [
  Array [
    Object {
      "countryId": "AR",
      "cpgId": "001",
      "eventDatetime": Any<String>,
      "eventInfo": "",
      "eventResult": "ERROR",
      "eventType": "RETRY",
      "orderId": 16,
      "organizationId": "3043",
    },
  ],
]
`;

exports[`Create Function You should create a log with status ERROR. 1`] = `
Array [
  Array [
    Object {
      "countryId": "AR",
      "cpgId": "001",
      "eventDatetime": Any<String>,
      "eventInfo": "Test Error",
      "eventResult": "ERROR",
      "eventType": "RETRY",
      "orderId": 16,
      "organizationId": "3043",
    },
  ],
]
`;

exports[`Create Function You should create a log with status OK. 1`] = `
Array [
  Array [
    Object {
      "countryId": "AR",
      "cpgId": "001",
      "eventDatetime": Any<String>,
      "eventInfo": "The operation has been successful. The order data was sent with id 16.",
      "eventResult": "OK",
      "eventType": "RETRY",
      "orderId": 16,
      "organizationId": "3043",
    },
  ],
]
`;
